reactioncommerce/reaction
# WIP: #4480 Convert product grid to consume GraphQL data

Resolves #4480  
Impact: **major**  
Type: **performance**

## Changes
This PR adds Apollo Client integration in our Meteor app, and allows the product grid to consume data via GraphQL. This is in progress, with some additions needed to the GraphQL API in order for the grid to be regression free (Search for TODO in the code).

## Breaking changes
TBD - depending on how custom shops have extended the grid, it may/may not break things.

## Testing
1. Load up the homepage
2. Confirm products load, infinite scroll works, clicking through to PDP - basically any and all functionality the Meteor grid has.
3. Confirm caching works - visit homepage, visit tag page, go back to homepage & confirm it loads instantly without another graphql network request
